### ISSUE TITLE
Update `fixed-point-wasm`

### ISSUE THREAD

**Issue by ryangoree** [delvtech/hyperdrive-frontend#1351]
Support for negative numbers just landed in delvtech/hyperdrive-rs#186. The wrappers should be updated to avoid bugs from accidentally passing a negative number to the lib and any special handling of negative numbers should be cleaned up. For example:
https://github.com/delvtech/hyperdrive-frontend/blob/9c7f6d7ea12c8862a957818fe8cea7dc258f6e5d/apps/hyperdrive-trading/src/base/formatRate.ts#L4-L10

**Comment by ryangoree** [comment]
Ran into a few gotchas integrating the generic FixedPoint type into the lib. One of them has been resolved via rustwasm/wasm-bindgen#4082, but I'm still working through the rest.

After some initial struggles with the bindings, I starting porting everything over to TypeScript and using `BigInt`s directly to avoid being limited by the largest signed type implemented in the `fixedpointmath`, `I256`. However, I realized the `ln` function is pretty cryptic and super optimized for efficiency and I think it would be better if both libs were built on the same source code.

I can make a single `ln` lib that the TypeScript port uses, but now I'm back to trying to make the bindings work, but with some new learnings gained while porting that will make it more efficient, easier to use, and slightly less bug prone.
